waku/v2/protocol/waku_noise/noise.nim

```diff
@@ -917,4 +923,565 @@ proc deserializePayloadV2*(payload: seq[byte]): Result[PayloadV2, cstring]
+
+      # Calls MixKey(DH(e, re)).
+      hs.ss.mixKey(dh(hs.e.privateKey, hs.re))
+
+    of T_es:
+
+      # If user is reading the "es" token
+
+      trace "noise dh es"
+
+      # We check if keys are correctly set.
+      # If both present, we call MixKey(DH(e, rs)) if initiator, MixKey(DH(s, re)) if responder.
+      if hs.initiator:
+        if hs.e == default(KeyPair) or hs.rs == default(Curve25519Key):
+          raise newException(NoisePublicKeyError, "Local or remote ephemeral/static key not set")
+        hs.ss.mixKey(dh(hs.e.privateKey, hs.rs))
+      else:
+        if hs.re == default(Curve25519Key) or hs.s == default(KeyPair):
+          raise newException(NoisePublicKeyError, "Local or remote ephemeral/static key not set")
+        hs.ss.mixKey(dh(hs.s.privateKey, hs.re))
+
+    of T_se:
+
+      # If user is reading the "se" token
+
+      trace "noise dh se"
+
+      # We check if keys are correctly set.
+      # If both present, call MixKey(DH(s, re)) if initiator, MixKey(DH(e, rs)) if responder.
+      if hs.initiator:
+        if hs.s == default(KeyPair) or hs.re == default(Curve25519Key):
+          raise newException(NoiseMalformedHandshake, "Local or remote ephemeral/static key not set")
+        hs.ss.mixKey(dh(hs.s.privateKey, hs.re))
+      else:
+        if hs.rs == default(Curve25519Key) or hs.e == default(KeyPair):
+          raise newException(NoiseMalformedHandshake, "Local or remote ephemeral/static key not set")
+        hs.ss.mixKey(dh(hs.e.privateKey, hs.rs))
+
+    of T_ss:
+
+      # If user is reading the "ss" token
+
+      trace "noise dh ss"
+
+      # If local and/or remote static keys are not set, we raise an error
+      if hs.s == default(KeyPair) or hs.rs == default(Curve25519Key):
+        raise newException(NoiseMalformedHandshake, "Local or remote static key not set")
+
+      # Calls MixKey(DH(s, rs)).
+      hs.ss.mixKey(dh(hs.s.privateKey, hs.rs))
+
+  return ok(outHandshakeMessage)
+
+#################################
+## Procedures to progress handshakes between users
+#################################
+
+# Initializes a Handshake State
+proc initialize*(hsPattern: HandshakePattern, staticKey: KeyPair = default(KeyPair), prologue: seq[byte] = @[], psk: seq[byte] = @[], preMessagePKs: seq[NoisePublicKey] = @[], initiator: bool = false): HandshakeState 
+  {.raises: [Defect, NoiseMalformedHandshake, NoiseHandshakeError, NoisePublicKeyError].} =
+  var hs = HandshakeState.init(hsPattern)
+  hs.ss.mixHash(prologue)
+  hs.s = staticKey
+  hs.psk = psk
+  hs.msgPatternIdx = 0
+  hs.initiator = initiator
+  # We process any eventual handshake pre-message pattern by processing pre-message public keys
+  processPreMessagePatternTokens(hs, preMessagePKs)
+  return hs
+
+# Advances 1 step in handshake
+proc stepHandshake*(rng: var BrHmacDrbgContext, hs: var HandshakeState, readPayloadV2: PayloadV2 = default(PayloadV2), transportMessage: seq[byte] = @[]): Result[HandshakeStepResult, cstring]
```

Comment:
As per our call, please add a comment about when `readPaylodV2` and `transportMessage`  should be passed.

Reply:
Addressed in https://github.com/status-im/nwaku/pull/934/commits/1d86c99113c3a79e07892e5cf8821baf104845a3